Mathlib/FieldTheory/KummerExtension.lean

```diff
@@ -0,0 +1,444 @@
+/-
+Copyright (c) 2023 Andrew Yang, Patrick Lutz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.RootsOfUnity.Basic
+import Mathlib.RingTheory.AdjoinRoot
+import Mathlib.LinearAlgebra.Charpoly.Basic
+import Mathlib.FieldTheory.Galois
+import Mathlib.LinearAlgebra.Eigenspace.Minpoly
+/-!
+# Kummer Extensions
+
+## Main result
+- `isCyclic_tfae`:
+Suppose `L/K` is a finite extension of dimension `n`, and `K` contains all `n`-th roots of unity.
+Then `L/K` is cyclic iff
+`L` is a splitting field of some irreducible polynomial of the form `Xⁿ - a : K[X]` iff
+`L = K[α]` for some `αⁿ ∈ K`.
+
+- `autEquivRootsOfUnity`:
+Given an instance `IsSplittingField K L (X ^ n - C a)`
+(perhaps via `isSplittingField_X_pow_sub_C_of_root_adjoin_eq_top`),
+then the galois group is isomorphic to `rootsOfUnity n K`, by sending
+`σ ↦ σ α / α` for `α ^ n = a`, and the inverse is given by `μ ↦ (α ↦ μ • α)`.
+
+- `autEquivZmod`:
+Furthermore, given an explicit choice `ζ` of a primitive `n`-th root of unity, the galois group is
+then isomorphic to `Multiplicative (ZMod n)` whose inverse is given by
+`i ↦ (α ↦ ζⁱ • α)`.
+
+## Other results
+Criteria for `X ^ n - C a` to be irreducible is given:
+`X_pow_sub_C_irreducible_iff_of_prime`: `X ^ n - C a` is irreducible iff `a` is not a `p`-power.
+
+TODO: criteria for general `n`.
```

Comment:
```suggestion
Criteria for `X ^ p - C a` to be irreducible is given for `p` prime:
`X_pow_sub_C_irreducible_iff_of_prime`: `X ^ p - C a` is irreducible iff `a` is not a `p`-power.

TODO: criteria for general `n`.
```
Maybe add a reference to Serge Lang's Algebra:
![image](https://github.com/leanprover-community/mathlib4/assets/3064145/f2a20314-bf00-4856-bf8f-80798c10f8bb)

Reply:
I was planning to list the criteria provided and hence the wording.